packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -78,6 +79,14 @@ export const plugin = <TContext>(
        */
       const logger = requestLogger || loggerForPlugin;
 
+      // If the options are false don't do any metrics timing.
+      if (options.traceReporting === false) {
+        metrics.captureTraces = false;
+        return {};
+      }
+
+      metrics.captureTraces = true;
```

Comment:
Can we just set this as part of shouldTraceOperation? I think setting it to true and later setting it to false is strange. (Setting to false above in the literal false case is fine.)